### DISCUSSION TITLE
Can the use of `IsNotNull` affect nullability checks?

### DISCUSSION THREAD

**Discussion by runenur** [thomhurst/TUnit#1212] [Ideas]
After I have used `IsNotNull` to assert that a variable is not `null`, I still have to treat it as if it could be `null` later on in the test:

```csharp
await Assert.That(result).IsNotNull();
await Assert.That(result?.Status).IsEqualTo("Completed");
```

The use of `?.` here feels unnecessary. Could it be annotated somehow? Or is it possible to use Roslyn analyzers to affect nullability checks?

**Comment by viceroypenguin** [comment]
~~@thomhurst if `[NotNull]` is added to the `value` parameter, then that will address this. NRT flow analysis expects that anything that is passed to a `[NotNull]` parameter is no longer null after that method returns.~~

**Reply by vbreuss** [reply]
But that would be incorrect, e.g.
```csharp 
object? sut = null;
await Assert.That(sut).IsNull();
```
Would tell the compiler that `sut` is not null

**Reply by viceroypenguin** [reply]
Ah, right.

**Reply by vbreuss** [reply]
The "Problem" is, that the actual check is in an extension method, without the subject as direct parameter.
